float64 ax
float64 ay
float64 ar
float64 T